src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+
+            # Calculate maximum distance to the object.
+            obj_max_distance = numpy.max(numpy.linalg.norm(obj.occupiedSpace.mesh.vertices - reg_candidate_point, axis=1))
+
+            if obj_max_distance > reg_circumradius:
+                return False
+
+        # PASS 5
+        # If the difference between the object's region and this region is empty,
+        # i.e. obj_region - self_region = EmptyRegion, that means the object is
+        # entirely contained in this region.
+        diff_region = obj.occupiedSpace.difference(self)
+
+        return isinstance(diff_region, EmptyRegion)
+
+    def containsRegionInner(self, reg, tolerance):
+        if tolerance != 0:
+            warnings.warn("Nonzero tolerances are ignored for containsRegionInner on MeshVolumeRegion")
+
+        if isinstance(reg, MeshVolumeRegion):
+            diff_region = reg.difference(self)
+
+            return isinstance(diff_region, EmptyRegion)
+
+        raise NotImplementedError
+
+    # Composition methods #
+    @lru_cache(maxsize=None)
+    def intersect(self, other, triedReversed=False):
+        """ Get a `Region` representing the intersection of this region with another.
+
+        This function handles intersection computation for `MeshVolumeRegion` with:
+        * `MeshVolumeRegion`
+        * `PolygonalFootprintRegion`
+        * `PolygonalRegion`
+        * `PathRegion`
+        * `PolylineRegion`
+        """
+        # If one of the regions isn't fixed fall back on default behavior
+        if needsSampling(self) or needsSampling(other):
+            return super().intersect(other, triedReversed)
+
+        if isinstance(other, MeshVolumeRegion):
+            # Other region is a mesh volume. We can extract the mesh to perform boolean operations on it
+            other_mesh = other.mesh
+
+            # Compute intersection using Trimesh (CalledProcessError usually means empty intersection for OpenSCAD)
+            try:
+                new_mesh = self.mesh.intersection(other_mesh, engine=self.engine)
+            except CalledProcessError:
+                if self.engine == "scad":
+                    return EmptyRegion("EmptyMesh")
+                else:
+                    raise RuntimeError("Mesh boolean operation failed.")
+            except ValueError as exc:
+                raise ValueError("Unable to compute mesh boolean operation. Do you have the Blender and OpenSCAD installed on your system?") from exc
+
+            if new_mesh.is_empty:
+                return EmptyRegion("EmptyMesh")
+            elif new_mesh.is_volume:
+                return MeshVolumeRegion(new_mesh, tolerance=min(self.tolerance, other.tolerance), centerMesh=False, engine=self.engine)
+            else:
+                # Something went wrong, abort
+                return super().intersect(other, triedReversed)
+
+        # WIP Code: Currently correct but VERY slow.
```

Comment:
Are we using this or not? Let's not leave it here but commented-out.

Reply:
Not currently in use, but it's what we would want to implement if we could get it faster. I can remove it if you'd like.

Reply:
If you want to keep the code around, please move it into a function; then you can just comment out the call to the function instead of the big block.

Reply:
I'll just remove it for now. We can always find it in the history of Scenic-devel if needed.